syntax = "proto3";

package greet;

service Greeter {
	// Unary streaming
	rpc SayHello (HelloRequest) returns (HelloReply);

	// Server Streaming
	rpc ServerSaysHello (HelloRequest) returns (stream HelloReply);

	// Client Streaming
	rpc ClientSaysHello (stream HelloRequest) returns (DelayedReply);

	// Bi-directional streaming
	rpc ClientServerChat (stream HelloRequest) returns (stream HelloReply);
}

message HelloRequest {
  string name = 1;
  string greeting = 2;
}

message HelloReply {
  string message = 1;
}

message DelayedReply {
	string message = 1;
	repeated HelloRequest request = 2;
}
